### DISCUSSION TITLE
Allow to copy the binary in memory inspector / ASCII memory view

### DISCUSSION THREAD

**Discussion by suXinjke** [RetroAchievements/RAIntegration#896] [Ideas]
The game I'm working on happens to involve strings in the memory. It's uncomfortable identifying them - I can't easily see what they are until I manually convert the binary into readable string, and to do that I also need to retype the binary manually.

I wish we could copy the binary in memory view, and I wish we had the ASCII on the right side, similar to many hex editors or cheat engine.

I heard newer generation games may involve such strings more often, so this might be real convenient.

**RAIntegration memory inspector:**
![image](https://user-images.githubusercontent.com/9765719/169313524-4eb6505d-db5e-455d-8fcc-e564e43c3fa3.png)

**HxD:**
![image](https://user-images.githubusercontent.com/9765719/169313554-d761ced2-c853-49ff-9688-beae43ffd809.png)

**Comment by Jamiras** [comment]
There is an "ASCII Text" search mode, which will decode the values for you.
![image](https://user-images.githubusercontent.com/32680403/169319917-ea2d59a2-9b47-43fe-9bee-c6e0d1547aa3.png)

You could force the behavior by doing an "ASCII Text" search from "0x1fe44cc-0x1fe44cc" where "equal last value" and use that.

I do plan to eventually allow "ASCII Text" as a data type in the bookmark window.

**Reply by suXinjke** [reply]
This is not helpful when you work with bigger amount of text rather than just PSX header

**Reply by Jamiras** [reply]
PSX header was just the quickest thing I had available for getting a screenshot.

My point is that there is a way to do this already, but I agree that there is room for improvement.

**Comment by Etron-RA** [comment]
An ASCII bookmark type would be sweet. I've been monitoring ASCII text with the RP, it's been pretty useful for Rez since the levels/models appear to be filenames (works well with the hacking theme of the game). Here's an example RP for reading the ASCII starting at 0x20c124:
`Swayzak is hacking @ASCIIChar(0xH20c124)@ASCIIChar(0xH20c125)@ASCIIChar(0xH20c126)@ASCIIChar(0xH20c127)@ASCIIChar(0xH20c128)@ASCIIChar(0xH20c129)@ASCIIChar(0xH20c12a)@ASCIIChar(0xH20c12b)@ASCIIChar(0xH20c12c)@ASCIIChar(0xH20c12d)@ASCIIChar(0xH20c12e)@ASCIIChar(0xH20c12f) system`

**Comment by Jamiras** [comment]
>  I wish we had the ASCII on the right side

Implemented in #1023
